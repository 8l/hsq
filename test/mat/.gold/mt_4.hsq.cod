	top:top top sqmain

.  _a:c1 _b:39 34 63 92 7 8 12 10 13 9 11 0

.  _c:10 120 1 49 7 56 15 49 0 0

_main:
	dec sp; ?+11; sp ?+7; ?+6; sp ?+2; 0
	?+6; sp ?+2; bp 0
	bp; sp bp
	dec sp; ?+11; sp ?+7; ?+6; sp ?+2; 0
	?+6; sp ?+2; t1 0
	dec sp; ?+11; sp ?+7; ?+6; sp ?+2; 0
	?+6; sp ?+2; t2 0
	dec sp; ?+11; sp ?+7; ?+6; sp ?+2; 0
	?+6; sp ?+2; t3 0

	t1; t2; c2 t1; c3 t1; t1 t2
	t1; t3; ?+11; t2 Z; Z ?+4; Z; 0 t1; t1 t3
	t2; t1; t3 t2; c4 t2; t2 t1
	t1 (-1)

	t1; t2; c2 t1; dec t1; t1 t2
	t1; t3; ?+11; t2 Z; Z ?+4; Z; 0 t1; t1 t3
	t2; t1; t3 t2; c4 t2; t2 t1
	t1 (-1)

	t1; t2; c2 t1; c5 t1; t1 t2
	t1; t3; ?+11; t2 Z; Z ?+4; Z; 0 t1; t1 t3
	t2; t1; t3 t2; c6 t2; t2 t1
	t1 (-1)

	t1; t2; c2 t1; c7 t1; t1 t2
	t1; t3; ?+11; t2 Z; Z ?+4; Z; 0 t1; t1 t3
	t2; t3 Z; Z t2; Z; c8 t2
	t2 (-1)

	t1; t2; c2 t1; c9 t1; t1 t2
	t1; t3; ?+11; t2 Z; Z ?+4; Z; 0 t1; t1 t3
	t2; t1; t3 t2; c10 t2; t2 t1
	t1 (-1)

	t1; t2; c2 t1; c11 t1; t1 t2
	t1; t3; ?+11; t2 Z; Z ?+4; Z; 0 t1; t1 t3
	t2; t1; t3 t2; c12 t2; t2 t1
	t1 (-1)

	t1; t2; c2 t1; c13 t1; t1 t2
	t1; t3; ?+11; t2 Z; Z ?+4; Z; 0 t1; t1 t3
	t2; t1; t3 t2; c12 t2; t2 t1
	t1 (-1)

	c8 (-1)

	t1; t2; c2 t1; c14 t1; t1 t2
	t1; t3; ?+11; t2 Z; Z ?+4; Z; 0 t1; t1 t3
	t2; t1; t3 t2; c4 t2; t2 t1
	t1 (-1)

	t1; t2; c2 t1; c15 t1; t1 t2
	t1; t3; ?+11; t2 Z; Z ?+4; Z; 0 t1; t1 t3
	t2; t1; t3 t2; c4 t2; t2 t1
	t1 (-1)

	t1; t2; c2 t1; c16 t1; t1 t2
	t1; t3; ?+11; t2 Z; Z ?+4; Z; 0 t1; t1 t3
	t2; t1; t3 t2; c4 t2; t2 t1
	t1 (-1)

	t1; t2; c2 t1; c8 t1; t1 t2
	t1; t3; ?+11; t2 Z; Z ?+4; Z; 0 t1; t1 t3
	t2; t1; t3 t2; c4 t2; t2 t1
	t1 (-1)

	c8 (-1)

	t1; t2; c17 t1; c3 t1; t1 t2
	t1; t3; ?+11; t2 Z; Z ?+4; Z; 0 t1; t1 t3
	t2; t1; t3 t2; c4 t2; t2 t1
	t1 (-1)

	t1; t2; c17 t1; dec t1; t1 t2
	t1; t3; ?+11; t2 Z; Z ?+4; Z; 0 t1; t1 t3
	t3 (-1)

	t1; t2; c17 t1; c5 t1; t1 t2
	t1; t3; ?+11; t2 Z; Z ?+4; Z; 0 t1; t1 t3
	t2; t1; t3 t2; c18 t2; t2 t1
	t1 (-1)

	t1; t2; c17 t1; c7 t1; t1 t2
	t1; t3; ?+11; t2 Z; Z ?+4; Z; 0 t1; t1 t3
	t3 (-1)

	t1; t2; c17 t1; c9 t1; t1 t2
	t1; t3; ?+11; t2 Z; Z ?+4; Z; 0 t1; t1 t3
	t2; t1; t3 t2; c19 t2; t2 t1
	t1 (-1)

	t1; t2; c17 t1; c11 t1; t1 t2
	t1; t3; ?+11; t2 Z; Z ?+4; Z; 0 t1; t1 t3
	t3 (-1)

	t1; t2; c17 t1; c13 t1; t1 t2
	t1; t3; ?+11; t2 Z; Z ?+4; Z; 0 t1; t1 t3
	t2; t1; t3 t2; c4 t2; t2 t1
	t1 (-1)

	t1; t2; c17 t1; c14 t1; t1 t2
	t1; t3; ?+11; t2 Z; Z ?+4; Z; 0 t1; t1 t3
	t3 (-1)

	t1; t2; c17 t1; c15 t1; t1 t2
	t1; t3; ?+11; t2 Z; Z ?+4; Z; 0 t1; t1 t3
	t2; t1; t3 t2; c10 t2; t2 t1
	t1 (-1)

	?+8; sp ?+4; t3; 0 t3; inc sp
	?+8; sp ?+4; t2; 0 t2; inc sp
	?+8; sp ?+4; t1; 0 t1; inc sp
	sp; bp sp
	?+8; sp ?+4; bp; 0 bp; inc sp
	?+8; sp ?+4; ?+7; 0 ?+3; Z Z 0

sqmain:
	dec sp; ?+11; sp ?+7; ?+6; sp ?+2; 0
	?+6; sp ?+2; ?+2 0 _main; . ?; inc sp

	Z Z (-1)

. c3:0 c8:10 c5:2 c6:20 c7:3 c9:4 c4:40 c19:48 c11:5 c10:50 c13:6 c12:60 c14:7 c15:8 c16:9 c18:97 c1:97 98 99 0 c2:_b c17:_c

. t1:0 t2:0 t3:0

. inc:-1 Z:0 dec:1 ax:0 bp:0 sp:-sp
