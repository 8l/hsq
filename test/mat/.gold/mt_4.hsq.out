OJSR9DH
2513
2xb178712